src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
+
+import { getAllUsers, getUserById } from '../../services/user';
+import { isValidLetter } from '../../services/Pattern';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  onAdd: (todo: Todo) => void;
+  newTodoId: number;
+}
+
+export const TodoForm: React.FC<Props> = ({ onAdd, newTodoId }) => {
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const TitleTetters = event.target.value;
+
+    if (isValidLetter(TitleTetters.slice(-1))) {
+      setTitle(TitleTetters);
+    } else {
+      setTitle(TitleTetters.slice(0, -1));
+    }
```

Comment:
this logic in not really clear. why do you need slice here?

Reply:
I did it to complete this task.
(* Optional) Allow entering only letters (ua and en), digits, and spaces in the title field. Just remove any other characters from the title.
Using isValidLetter(TitleTetters.slice(-1)), I check the last added character in the string, and if it matches the pattern is Valid,i use Titleletters. If it doesn't match, I shorten the string using setTitle(TitleTetters.slice(0, -1)), thereby removing the last character from the string.